app/src/main/admin/reducers/modellingGroupsReducer.ts

```diff
@@ -0,0 +1,31 @@
+import { ModellingGroupsAction, ModellingGroupTypes } from "../actionTypes/ModellingGroupsTypes";
+import {ModellingGroup, ModellingGroupDetails, User} from "../../shared/models/Generated";
+
+export interface ModellingGroupsState {
+    groups: ModellingGroup[];
+    currentGroup: ModellingGroup;
+    currentGroupDetails: ModellingGroupDetails;
+    currentGroupMembers: User[];
+}
+
+export const modellingGroupInitialState: ModellingGroupsState = {
+    groups: [],
+    currentGroup: null,
+    currentGroupDetails: null,
+    currentGroupMembers: []
+};
+
+export const modellingGroupsReducer = (state = modellingGroupInitialState, action: ModellingGroupsAction) => {
+    switch (action.type) {
+        case ModellingGroupTypes.ADMIN_GROUPS_FETCHED:
+            return {...state, groups: action.data ? action.data : [] };
```

Comment:
Why do we need this guard to make sure it's an array?

Reply:
just to keep up with initial data format of reducers, for example we have some list , and by some reason instead of array some null or undefined was passed to it, that will help keeping , it as empty array. It's not "must" logic, it can be different here, that is just something i have chosen to minimise possible errors.

Reply:
but here i was only checking if not empty, maybe i should check if array too